lua/entities/gmod_wire_expression2/core/cl_http_cl.lua

```diff
@@ -0,0 +1,178 @@
+--[[
+
+	Clientside HTTP library
+	 - swadical 🍌
+
+]]
+
+--	Clientside HTTP is separate from serverside HTTP:
+--		Serverside HTTP allows greater speeds than clientside HTTP and it is
+--		up to the server owner to decide which extension should be enabled
+--		through a convar (found in core/http.lua)
+
+--	This library will mirror the functionality of core/http.lua as best as
+--	possible, making sure any existing e2s which make use of the http library
+--	are not broken by this extension
+
+E2Lib.clHTTP = {}
+local lib = E2Lib.clHTTP
+
+lib.netMsgID = "wire_expression2_cl_http" -- open to change
+lib.whitelistSyncNetMsgID = "wire_expression2_cl_http_whitelist" -- open to change
+lib.currentUID = 0
+lib.defaultWhitelistData = [[#
+#	The Wiremod Expression 2 client-side HTTP request whitelist
+#
+#	Hints:
+#	- One entry per line!
+#	- Entries are also lua string patterns
+#	- Anything after a # in an entry is ignored and treated as a comment
+#	- You do not have to have a new line at the end of this file.
+#	- Empty lines are ignored
+#
+#	Have fun!
+
+#^https?://www.github.com/.*$
+#^https?://www.github.com/.*$
+]]
+
+-- number function(void)
+local function newUID()
+	lib.currentUID = (lib.currentUID + 1) % 255
+	return lib.currentUID -- return a number between 0 and 254
+end
+
+lib.rawRequest = http.Fetch
+lib.requestMaxCompressedSize = 16 * 1024 * 1024 -- 16 megabytes
+lib.segmentSize = 4 * 1024-- 4 kilobytes
+lib.segmentBits = math.min(math.ceil(math.log(lib.segmentSize,2)) + 2,32)
+lib.sendInterval = 0.5-- seconds
+lib.HTTP_REQUEST_FAILED = "did not receive HTTP body" -- message for HTTP failure
+lib.HTTP_REQUEST_TOO_BIG = "request was too big to handle" -- message for when a request is too large
+
+-- void function(string id, number interval, function func)
+-- Internal: Do not call.
+local function timerCoroutine(id,interval,func)
+	func = coroutine.wrap(func)
+	timer.Create(id,interval,0,function()
+		if func() then
+			timer.Remove(id)
+		end
+	end)
+end
+
+-- void function(number uid, string body_compressed, number length)
+-- Internal: Do not call.
+local function writeHTTPBody(uid,body,length)
+	local segments = math.ceil(length/lib.segmentSize)
+
+	-- reliable netmessages are always sent in order
+	for i=1,segments do
+		local segment = body:sub(lib.segmentSize * (i-1) + 1,lib.segmentSize * i)
+		net.Start(lib.netMsgID)
+			net.WriteUInt(uid,8)						-- this is who we are
+			net.WriteBool(false)						-- No; the request did not error
+			net.WriteBool(true)							-- yes we are writing a body
+			net.WriteUInt(#segment,lib.segmentBits)	-- the body is this much long
+			net.WriteData(segment,#segment)				-- and here is the body
+		net.SendToServer()
+
+		coroutine.yield()
+	end
+end
+
+-- void function(number uid, number length, table headers, number code)
+-- Internal: Do not call.
+local function writeMetadata(uid,length,headers,code)
+	net.Start(lib.netMsgID)
+		net.WriteUInt(uid,8)						-- this is who we are
```

Comment:
I suggest putting these comments on the line before the one they're referring to